amm/contracts/stable_pool/lib.rs

```diff
@@ -582,14 +615,19 @@ pub mod stable_pool {
                 shares: shares_to_burn,
                 to,
             });
+            self.env().emit_event(Sync {
+                reserves: self.reserves(),
+            });
             Ok((shares_to_burn, fee_part))
         }
 
         #[ink(message)]
         fn force_update_rate(&mut self) {
             let current_time = self.env().block_timestamp();
-            for rate in self.pool.token_rates.iter_mut() {
-                rate.update_rate_no_cache(current_time);
+            for (id, rate) in self.pool.token_rates.iter_mut().enumerate() {
+                if rate.update_rate_no_cache(current_time) {
+                    Self::env().emit_event(TokenRateUpdated { token: self.pool.tokens[id], rate: *rate })
```

Comment:
How about we emit single event with all the rates updated at the end? So, if any of the rates gets updated, we set some mutable flag, and emit one event at the end with currently-known rates.

Reply:
This way we would include rates for non-rated tokens (which is constant) but in the case of a pool with more than one rated token it would limit the number of emitted events to one. So that's the trade-off. I think it can be done the way you are suggesting :+1:

Reply:
I think it's fine to include the rates of not-updated tokens. From the pov of an indexer that will be easier to integrate as well.